### PR TITLE
added sql-data-types branch and corresponding doc

### PR DESCRIPTION
**Pull request by lwandzura**
Please review the sql data types doc and let me know what needs to be added and/or removed.

### CODE REVIEW

File: content/influxdb/cloud-iox/sql-data-types.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
+---
+title: SQL data types
+description: >
+  SQL data types
```
Comment (sanderson) [inline content/influxdb/cloud-iox/sql-data-types.md:4]:
```suggestion
  The InfluxDB SQL implementation supports a number of data types including 64-bit integers,
  double-precision floating point numbers, strings, and more.
```

File: content/influxdb/cloud-iox/sql-data-types.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+menu:
+  influxdb_cloud_iox:
+    name: SQL data types
+    parent: Query data with SQL
+weight: 105
+---
+
+InfluxDB Cloud backed by InfluxDB IOx uses the Apache Arrow DataFusion implementation of SQL. Data types define the type of value that can be stored in table column. In InfluxDB's SQL implementation, a `measurement` is a table, and columns are comprised of `tags`, `fields` and `timestamps`.
```
Comment (sanderson) [inline content/influxdb/cloud-iox/sql-data-types.md:12]:
```suggestion
InfluxDB Cloud backed by InfluxDB IOx uses the [Apache Arrow DataFusion](https://arrow.apache.org/datafusion/) implementation of SQL.
Data types define the type of values that can be stored in table columns.
In InfluxDB's SQL implementation, a **measurement** is structured as a table,
and columns are comprised of **tags**, **fields** and **timestamps**. 
```

File: content/influxdb/cloud-iox/sql-data-types.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+menu:
+  influxdb_cloud_iox:
+    name: SQL data types
+    parent: Query data with SQL
+weight: 105
+---
+
+InfluxDB Cloud backed by InfluxDB IOx uses the Apache Arrow DataFusion implementation of SQL. Data types define the type of value that can be stored in table column. In InfluxDB's SQL implementation, a `measurement` is a table, and columns are comprised of `tags`, `fields` and `timestamps`. 
+
+DataFusion uses the Arrow type system for query execution. Data types stored in InfluxDB's storage engine are mapped to SQL data types according to the following tables.
```
Comment (sanderson) [inline content/influxdb/cloud-iox/sql-data-types.md:14]:
```suggestion
DataFusion uses the Arrow type system for query execution.
Data types stored in InfluxDB's storage engine are mapped to SQL data types at query time. 
```

File: content/influxdb/cloud-iox/sql-data-types.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+{{% note %}}
+When performing casting operations, cast to the **name** of the data type, not the actual data type.  Names are not case sensitive. 
+
+```sql
+--Examples
+SELECT
+'99'::BIGINT
+'2019-09-18T00:00:00Z'::timestamp
+```
```
Comment (sanderson) [inline content/influxdb/cloud-iox/sql-data-types.md:24]:
~~~suggestion
When performing casting operations, cast to the **name** of the data type, not the actual data type. 
Names are _case-insensitive_. For example:

```sql
SELECT
  '99'::BIGINT, 
  '2019-09-18T00:00:00Z'::timestamp
```
~~~

File: content/influxdb/cloud-iox/sql-data-types.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+| CHAR    | UTF8      | Character string, fixed-length    |
+| VARCHAR | UTF8      | Character string, variable-length |
+| TEXT    | UTF8      | Variable unlimited length         |
+
+```sql
+--Examples
+abcdefghijk
+time
+"h2o_temperature"
+```
```
Comment (sanderson) [inline content/influxdb/cloud-iox/sql-data-types.md:40]:
~~~suggestion
##### Example character types

```sql
abcdefghijk
time
"h2o_temperature"
```
~~~

File: content/influxdb/cloud-iox/sql-data-types.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+## Numeric types
+
+InfluxDB stores all integers as signed 64bit integers. The following numeric types are supported:
```
Comment (sanderson) [inline content/influxdb/cloud-iox/sql-data-types.md:44]: InfluxDB supports both signed and unsigned integers.

```suggestion
The following numeric types are supported:
```

File: content/influxdb/cloud-iox/sql-data-types.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+| Name            | Data type | Description                  |
+| :-------------- | :-------- | :--------------------------- |
+| BIGINT          | INT64     | large-range integer          |
+| BIGINT UNSIGNED | INT64     | large-range unsigned integer |
+| DOUBLE          | FLOAT64   | 64-bit floating-point number |
```
Comment (sanderson) [inline content/influxdb/cloud-iox/sql-data-types.md:50]:
```suggestion
| BIGINT          | INT64     | 64-bit signed integer        |
| BIGINT UNSIGNED | INT64     | 64-bit unsigned integer      |
| DOUBLE          | FLOAT64   | 64-bit floating-point number |
```

File: content/influxdb/cloud-iox/sql-data-types.md [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+
+
+Minimum integer:` -9223372036854775808`  
+Maximum integer: `9223372036854775807`
```
Comment (sanderson) [inline content/influxdb/cloud-iox/sql-data-types.md:54]:
```suggestion
Minimum signed integer:` -9223372036854775808`  
Maximum signed integer: `9223372036854775807`
```

File: content/influxdb/cloud-iox/sql-data-types.md [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+
+Minimum unsigned integer (uinteger): `0`  
+Maximum unsigned integer (uinteger): `18446744073709551615`  
+
+Floats can be a decimal point, decimal integer or decimal fraction.
```
Comment (sanderson) [inline content/influxdb/cloud-iox/sql-data-types.md:59]:
```suggestion
Floats can be a decimal point, decimal integer, or decimal fraction.
```

File: content/influxdb/cloud-iox/sql-data-types.md [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+
+```sql
+--Examples
+23.8
+-446.89
+5.00
+0.033
+```
```
Comment (sanderson) [inline content/influxdb/cloud-iox/sql-data-types.md:67]:
~~~suggestion
##### Example float types

```sql
23.8
-446.89
5.00
0.033
```
~~~

File: content/influxdb/cloud-iox/sql-data-types.md [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+
+## Time types
+
+A time type is a single point in time using nanosecond precision.  
+
+| Name      | Data type | Description                |
+| :-------- | :-------- | :------------------------- |
+| TIMESTAMP | TIMESTAMP | TimeUnit::Nanosecond, None |
+
+
```
Comment (sanderson) [inline content/influxdb/cloud-iox/sql-data-types.md:77]:
```suggestion
```

File: content/influxdb/cloud-iox/sql-data-types.md [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+```sql
+--Examples
+YYYY-MM-DDT00:00:00.000Z 
+YYYY-MM-DDT00:00:00.000-00:00 
+YYYY-MM-DD 00:00:00.000-00:00 
+YYYY-MM-DDT00:00:00Z
+YYYY-MM-DD 00:00:00.000
+YYYY-MM-DD 00:00:00
+1567296000000000000 
+1566176400
```
Comment (sanderson) [inline content/influxdb/cloud-iox/sql-data-types.md:89]: We need to be careful here. These aren't time types. These are integers. You can cast integers to a TIMESTAMP, but they aren't natively recognized as timestamps.

```suggestion
```

File: content/influxdb/cloud-iox/sql-data-types.md [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+| Name    | Data type | Description                                                           |
+| :------ | :-------- | :-------------------------------------------------------------------- |
+| BOOLEAN | BOOLEAN   | TRUE or FALSE for strings, 0 and 1 for integers, uintegers and floats |
+
+Booleans are parsed in the following manner:
+
+- string: `TRUE` or `FALSE`
+- integer: 0 is false, 1 is true
+- uinteger: 0 is false, 1 is true
+- float: 0.0 is false, 1.0 is true
```
Comment (sanderson) [inline content/influxdb/cloud-iox/sql-data-types.md:105]:
```suggestion
- integer: 0 is false, non-zero is true
- uinteger: 0 is false, non-zero is true
- float: 0.0 is false, non-zero is true
```

File: content/influxdb/cloud-iox/sql-data-types.md [hunk 14]
```diff
@@ -0,0 +1,133 @@
...
+
+Note that zero vaules are parsed as `false` and non zero negative values are parsed as `true`.
```
Comment (sanderson) [inline content/influxdb/cloud-iox/sql-data-types.md:107]: This is redundant with the change above.

```suggestion
```

File: content/influxdb/cloud-iox/sql-data-types.md [hunk 15]
```diff
@@ -0,0 +1,133 @@
...
+
+Note that zero vaules are parsed as `false` and non zero negative values are parsed as `true`. 
+
+```sql
+--Examples
+true
+TRUE
+false
+FALSE
+```
```
Comment (sanderson) [inline content/influxdb/cloud-iox/sql-data-types.md:115]:
~~~suggestion
##### Example boolean types

```sql
true
TRUE
false
FALSE
```
~~~

File: content/influxdb/cloud-iox/sql-data-types.md [hunk 16]
```diff
@@ -0,0 +1,133 @@
...
+- VARBINARY
+- REGCLASS
+- NVARCHAR
+- CUSTOM
+- ARRAY
+- ENUM
+- SET
+- INTERVAL
+- DATETIME
+- BYTEA
```
Comment (sanderson) [inline content/influxdb/cloud-iox/sql-data-types.md:133]: We should probably include all the non-64-bit numeric types that the InfluxDB doesn't support in this list as well.

```suggestion
## Unsupported SQL types

The following SQL types are not currently supported:

- UUID
- BLOB
- CLOB
- BINARY
- VARBINARY
- REGCLASS
- NVARCHAR
- CUSTOM
- ARRAY
- ENUM
- SET
- INTERVAL
- DATETIME
- BYTEA
```

**Review by sanderson — CHANGES_REQUESTED** [review]

**Review by sanderson — APPROVED** [review]

**Review by alamb — COMMENTED** [review]
This is sweet -- thank you @lwandzura

File: content/influxdb/cloud-iox/sql-data-types.md [hunk 17]
```diff
@@ -0,0 +1,137 @@
...
+- CLOB
+- BINARY
+- VARBINARY
+- REGCLASS
+- NVARCHAR
+- CUSTOM
+- ARRAY
+- ENUM
+- SET
+- INTERVAL
```
[the remark this answers is not in the document]
Comment (sanderson) [reply content/influxdb/cloud-iox/sql-data-types.md:135]: @alamb Thanks for calling this out. Do you know the explicit interval literal formats we support? Are they the same as or similar to those listed [here](https://media.datadirect.com/download/docs/openaccess/alloa/index.html#page/sqlref/interval-literals.html)?

[the remark this answers is not in the document]
Comment (alamb) [reply content/influxdb/cloud-iox/sql-data-types.md:135]: The logical types supported by arrow (and thus by DataFusion) are listed (somewhat obscurely) in https://docs.rs/arrow/31.0.0/arrow/datatypes/index.html

So in this case the interval types are: 

[IntervalDayTimeType](https://docs.rs/arrow/31.0.0/arrow/datatypes/struct.IntervalDayTimeType.html)	A “calendar” interval type in days and milliseconds.
[IntervalMonthDayNanoType](https://docs.rs/arrow/31.0.0/arrow/datatypes/struct.IntervalMonthDayNanoType.html)	A “calendar” interval type in months, days, and nanoseconds.
[IntervalYearMonthType](https://docs.rs/arrow/31.0.0/arrow/datatypes/struct.IntervalYearMonthType.html)	A “calendar” interval type in months.

That is not quite the same as the SQL equivalent. Here are the types when writing about intervals:


```sql
❯ select arrow_typeof(interval '1 month');
+-------------------------------------+
| arrowtypeof(IntervalYearMonth("1")) |
+-------------------------------------+
| Interval(YearMonth)                 |
+-------------------------------------+
1 row in set. Query took 0.006 seconds.
❯ select arrow_typeof(interval '1 second');
+--------------------------------------+
| arrowtypeof(IntervalDayTime("1000")) |
+--------------------------------------+
| Interval(DayTime)                    |
+--------------------------------------+
1 row in set. Query took 0.000 seconds.
```

File: content/influxdb/cloud-iox/sql-data-types.md [hunk 18]
```diff
@@ -0,0 +1,137 @@
...
+    parent: Query data with SQL
+weight: 105
+---
+
+InfluxDB Cloud backed by InfluxDB IOx uses the [Apache Arrow DataFusion](https://arrow.apache.org/datafusion/) implementation of SQL.
+Data types define the type of values that can be stored in table columns.
+In InfluxDB's SQL implementation, a **measurement** is structured as a table,
+and columns are comprised of **tags**, **fields** and **timestamps**. 
+
+DataFusion uses the Arrow type system for query execution.
```
[the remark this answers is not in the document]
Comment (lwandzura) [reply content/influxdb/cloud-iox/sql-data-types.md:18]: done

File: content/influxdb/cloud-iox/sql-data-types.md (hunk 17 above)
[the remark this answers is not in the document]
Comment (lwandzura) [reply content/influxdb/cloud-iox/sql-data-types.md:135]: added interval and examples

**Pull request merged**